src/ensembl/src/content/app/browser/track-panel/trackPanelActions.ts

```diff
@@ -99,10 +106,62 @@ export const changeTrackPanelModalViewForGenome: ActionCreator<
   );
 };
 
+export const updatePreviouslyViewedObjectsAndSave: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = () => (dispatch, getState: () => RootState) => {
+  const state = getState();
+  const activeGenomeId = getBrowserActiveGenomeId(state);
+  const activeEnsObject = getBrowserActiveEnsObject(state);
+  if (!activeGenomeId || !activeEnsObject) {
+    return;
+  }
+  const trackStates = getBrowserTrackStates(state)[activeGenomeId];
+
+  const previouslyViewedObjects = [
+    ...getActiveGenomePreviouslyViewedObjects(state)
+  ];
+
+  const existingIndex = previouslyViewedObjects.findIndex(
+    (previouslyViewedObject) =>
+      previouslyViewedObject.object_id === activeEnsObject.object_id
+  );
+  if (existingIndex === -1) {
+    // IF it is not present, add it to the end
+    previouslyViewedObjects.push({
+      genome_id: activeEnsObject.genome_id,
+      object_id: activeEnsObject.object_id,
+      object_type: activeEnsObject.object_type,
+      label: activeEnsObject.label,
+      trackStates: { ...trackStates }
+    });
```

Comment:
So, we are creating an infinite history of previously viewed objects... Is this _really_ what we want to do? I know Andrea says that we don’t know how many we want until we try, but realistically, how many previously viewed objects should we allow our application to store? A hundred? A thousand? Ten thousand? Where is the threshold after which they have definitely become no more than useless noise?